build/demo-launcher/src/launcher/main.py

```diff
@@ -264,10 +265,13 @@ async def boot(scheduler: Scheduler):
         sshkey_private = asyncssh.import_private_key(
             credentials.private_key, passphrase=credentials.passphrase
         )
+        sshkey_cert_public = ()
```

Comment:
Just to understand better, you are initializing `sshkey_cert_public` as an empty `tuple` so in case `credentials.public_cert` is empty, it doesn't raise an error on `sshClient()`? .
Looks that before this might bring an exception.

Did I get it right?

Reply:
exactly!
The issue was that if credentials.public_cert was empty we were accessing an undefined variable.
sshkey_cert_public needs to be always defined and with a value that is compatible with SSHClientConnectionOptions